### PR TITLE
[16.0] fastapi: eases tests development

### PR DESCRIPTION
**Pull request by lmignon**
@marielejeune @sbidoul @AnizR This PR provides a base class to ease the development of your tests. Your feedback will be welcome

### CODE REVIEW

File: fastapi/README.rst [hunk 1]
```diff
@@ -1006,49 +1006,63 @@ normally provided by the normal processing of the request by the fastapi app.
...
 
-  from functools import partial
+  from odoo.fastapi.tests.common import FastAPITransactionCase
 
-  from requests import Response
+  from odoo.addons.fastapi import dependencies
+  from odoo.addons.fastapi.routers import demo_router
 
-  from odoo.tests.common import TransactionCase
+  class FastAPIDemoCase(FastAPITransactionCase):
```
Comment (AnizR) [inline fastapi/README.rst:1028]: I think that we should run the test with @tagged("post_install", "-at_install") or did it change?

**Review by AnizR — CHANGES_REQUESTED** [review]

File: fastapi/README.rst [hunk 2]
```diff
@@ -1006,49 +1006,63 @@ normally provided by the normal processing of the request by the fastapi app.
...
+route handlers. This method is designed to be used when you need to test your
+app or when you need to test a specific router (It's therefore easy to defines
+tests for routers in an addon that doesn't provide a fastapi endpoint).
+
+With this base class, writing a test for a route handler is as simple as:
 
 .. code-block:: python
 
-  from functools import partial
+  from odoo.fastapi.tests.common import FastAPITransactionCase
```
Comment (marielejeune) [inline fastapi/README.rst:1023]:
```suggestion
  from odoo.addons.fastapi.tests.common import FastAPITransactionCase
```

File: fastapi/readme/USAGE.rst [hunk 3]
```diff
@@ -931,49 +931,63 @@ normally provided by the normal processing of the request by the fastapi app.
 to test the behavior of your route handlers when the partner is not authenticated,
 you can also inject a mock for the odoo_env etc...)
 
-With all these features, writing a test for the 'Hello world' route handler
-defined into the demo app is as simple as
+The fastapi addon provides a base class for the test cases that you can use to
+write your tests. This base class is **'odoo.fastapi.tests.common.FastAPITransactionCase'**.
```
Comment (marielejeune) [inline fastapi/readme/USAGE.rst:946]:
```suggestion
write your tests. This base class is **'odoo.addons.fastapi.tests.common.FastAPITransactionCase'**.
```

File: fastapi/readme/USAGE.rst [hunk 4]
```diff
@@ -931,49 +931,63 @@ normally provided by the normal processing of the request by the fastapi app.
...
+route handlers. This method is designed to be used when you need to test your
+app or when you need to test a specific router (It's therefore easy to defines
+tests for routers in an addon that doesn't provide a fastapi endpoint).
+
+With this base class, writing a test for a route handler is as simple as:
 
 .. code-block:: python
 
-  from functools import partial
+  from odoo.fastapi.tests.common import FastAPITransactionCase
```
Comment (marielejeune) [inline fastapi/readme/USAGE.rst:959]:
```suggestion
  from odoo.addons.fastapi.tests.common import FastAPITransactionCase
```

File: fastapi/readme/USAGE.rst [hunk 5]
```diff
@@ -931,49 +931,63 @@ normally provided by the normal processing of the request by the fastapi app.
...
-          )
-          cls.client = TestClient(cls.app)
-          cls._ctx_token = odoo_env_ctx.set(cls.env)
+.. code-block:: python
 
-      @classmethod
-      def tearDownClass(cls) -> None:
-          odoo_env_ctx.reset(cls._ctx_token)
-          cls.fastapi_demo_app._reset_app()
+  from odoo.fastapi.tests.common import FastAPITransactionCase
```
Comment (marielejeune) [inline fastapi/readme/USAGE.rst:985]:
```suggestion
  from odoo.addons.fastapi.tests.common import FastAPITransactionCase
```

**Review by AnizR — APPROVED** [review]
Code LGTM, I will try it on my module

File: extendable_fastapi/tests/common.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.init_extendable_registry()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        super().tearDownClass()
+        cls.reset_extendable_registry()
```
Comment (sbidoul) [inline extendable_fastapi/tests/common.py]:
```suggestion
        cls.reset_extendable_registry()
        super().tearDownClass()
```

File: base_rest_demo/tests/common.py [hunk 7]
```diff
@@ -29,14 +29,18 @@ def setUpClass(cls):
         cls.new_api_services_env = WorkContext(
             model_name="rest.service.registration", collection=collection
         )
-        cls.setUpExtendable()
+        cls.init_extendable_registry()
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        cls.reset_extendable_registry()
```
Comment (sbidoul) [inline base_rest_demo/tests/common.py]:
```suggestion
        cls.reset_extendable_registry()
        super().tearDownClass()
```

to tear down in reverse order as setup?

Comment (lmignon) [reply base_rest_demo/tests/common.py]: done

File: extendable_fastapi/tests/common.py (hunk 6 above)
Comment (lmignon) [reply extendable_fastapi/tests/common.py]: done

**Comment by lmignon** [comment]
Fixed and rebased after merge of #353

File: fastapi/readme/USAGE.rst [hunk 8]
```diff
@@ -527,7 +538,7 @@ current request.
         dependencies=[Depends(authenticated_partner)],
     )
     async def endpoint_app_info(
-        endpoint: FastapiEndpoint = Depends(fastapi_endpoint),  # noqa: B008
+        endpoint: Annotated[FastapiEndpoint, Depends(fastapi_endpoint)],  # noqa: B008
```
Comment (FrancoMaxime) [inline fastapi/readme/USAGE.rst]: Is the # noqa: B008 still required? 
It is not used in other function declarations below

**Review by FrancoMaxime — COMMENTED** [review]
LGTM

**Pull request merged**